### PR TITLE
Use local pre commit hooks

### PR DESCRIPTION
**Pull request by huppd**
changes the pre-commit hooks to local repo, such that the development and CI is using the same versions.

### CODE REVIEW

**Comment by huppd** [comment]
launch jenkins

**Comment by huppd** [comment]
launch jenkins

File: tmpl/.pre-commit-config.yaml [hunk 1]
```diff
@@ -1,21 +1,67 @@
 repos:
--   repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v4.3.0
+-   repo: local
+    # Running hooks locally provides them access the dependencies
     hooks:
     -   id: check-ast
```
Comment (ruestefa) [inline tmpl/.pre-commit-config.yaml:5]: IMHO running these checkers locally is overkill, as they are only used by pre-commit. It only makes sense for tools that are in `requirements.yml`, like `flake8` and `black`.

File: tmpl/requirements/dev-environment.yml [hunk 2]
```diff
@@ -74,6 +74,7 @@ dependencies:
   - platformdirs=2.5.1=pyhd8ed1ab_0
   - pluggy=1.0.0=py310hff52083_3
   - pre-commit=2.19.0=py310hff52083_0
+  - pre-commit-hooks=4.4.0=pyhd8ed1ab_0
```
Comment (ruestefa) [inline tmpl/requirements/dev-environment.yml:77]: See comment above.

**Comment by huppd** [comment]
launch jenkins

File: tmpl/.pre-commit-config.yaml (hunk 1 above)
Comment (huppd) [reply tmpl/.pre-commit-config.yaml:5]: I see your point and will happily change that back. I only see a small advantage adding `pre-commit-hooks` to the `dev-requirements.yml`, which is that the version control of all the hooks happens at exactly one place and not two.

**Comment by huppd** [comment]
launch jenkins

File: tmpl/.pre-commit-config.yaml (hunk 1 above)
Comment (regDaniel) [reply tmpl/.pre-commit-config.yaml]: To be honest, I think for these hooks it probably doesn't really matter. So no strong opinion here. If you are both (@huppd and @ruestefa ) fine with the latest changes, we can merge.

Comment (regDaniel) [reply tmpl/.pre-commit-config.yaml]: on second thought, if we make some hooks local, we should do it for all of them no? And we have these tools in `requirements.yml` if we add `pre-commit-hooks`. So the disadvantage is one additional dependency right? Anyways I'm fine with merging either way, just curious if I am missing a point here.

Comment (ruestefa) [reply tmpl/.pre-commit-config.yaml]: If it's possible to run these hooks locally without having to expand each entry from typically one line to a multiline block in `.pre-commit-config.yml`, then either way is fine with me. However, if they need to be specified as verbosely as in the original PR, than that IMHO blows up the file too much for pretty much zero gain as the hooks are never run outside pre-commit and are generally simple enough that updating them is not that crucial.

Comment (regDaniel) [reply tmpl/.pre-commit-config.yaml]: yep ok I'm still very indifferent but I see the point. I'll merge as is.

Comment (ruestefa) [reply tmpl/.pre-commit-config.yaml]: But I'm not going to stand in the way if you decide on running them locally. :)

**Pull request merged**